tests/framework/extensions/workloads/pods/pod_status.go

```diff
@@ -17,86 +16,77 @@ const (
 
 // StatusPods is a helper function that uses the steve client to list pods on a namespace for a specific cluster
 // and return the statuses in a list of strings
-func StatusPods(client *rancher.Client, clusterID string) ([]string, []error) {
+func StatusPods(client *rancher.Client, clusterID string) []error {
 	downstreamClient, err := client.Steve.ProxyDownstream(clusterID)
 	if err != nil {
-		return nil, []error{err}
+		return []error{err}
 	}
 
-	var podResults []string
 	var podErrors []error
 
 	steveClient := downstreamClient.SteveType(PodResourceSteveType)
 	err = kwait.Poll(5*time.Second, 15*time.Minute, func() (done bool, err error) {
-		podResults = []string{}
+		// emptying pod errors every time we poll so that we don't return stale errors
 		podErrors = []error{}
 
 		pods, err := steveClient.List(nil)
 		if err != nil {
-			podErrors = append(podErrors, err)
+			// not returning the error in this case, as it could cause a false positive if we start polling too early.
 			return false, nil
 		}
 
-		podResults = append(podResults, "Pod's Status: \n")
-
 		for _, pod := range pods.Data {
-			podResult, podError, err := CheckPodStatus(&pod)
-			if err != nil {
+			isReady, err := IsPodReady(&pod)
+			if !isReady {
 				// not returning the error in this case, as it could cause a false positive if we start polling too early.
 				return false, nil
 			}
-			if podError != nil {
-				podErrors = append(podErrors, podError)
-				return false, nil
-			}
-			if podResult != "" {
-				podResults = append(podResults, podResult)
-			}
-		}
 
-		if len(podResults) > 0 && len(podErrors) == 0 {
-			return true, nil
+			if err != nil {
+				podErrors = append(podErrors, err)
+			}
 		}
-
-		return false, nil
+		return true, nil
 	})
 
 	if err != nil {
-		return nil, []error{err}
+		return []error{err}
```

Comment:
I think @igomez06 meant to append this error too. So let's say one of the pods is in an error state, and the poll timed out because of that. This error check is for the poll, so it won't reach down below the line to return pod errors. That's why I think we need to append this returned poll error (which is `err`) with the pod errors (`podErrors`). That way, if the poll has an error, we can still know why x pod failed with an error y.

Reply:
I see, so as written this would only return 1 of the possibly `n` errors for polling. Let me update that